src/cargo_communication/request_handle.rs

```diff
@@ -0,0 +1,48 @@
+use crate::bsp_types::requests::Request;
+use crate::cargo_communication::cargo_command::CreateCommand;
+use crate::cargo_communication::cargo_result::CargoResult;
+use crate::cargo_communication::event::Event;
+use crate::cargo_communication::request_actor::RequestActor;
+use crate::communication::{Message, RequestId};
+use crossbeam_channel::{unbounded, Sender};
+use std::path::Path;
+
+#[derive(Debug)]
+pub struct RequestHandle {
+    sender_to_cancel: Sender<Event>,
+    _thread: jod_thread::JoinHandle,
+}
+
+impl RequestHandle {
+    pub fn spawn<R>(
+        sender_to_main: Box<dyn Fn(Message) + Send>,
+        req_id: RequestId,
+        params: R::Params,
+        root_path: &Path,
+    ) -> RequestHandle
+    where
+        R: Request + 'static,
+        R::Params: CreateCommand + Send,
+        R::Result: CargoResult,
+    {
+        let mut actor: RequestActor<R> =
+            RequestActor::new(sender_to_main, req_id, params, root_path);
+        let (sender_to_cancel, receiver_to_cancel) = unbounded::<Event>();
+        let thread = jod_thread::Builder::new()
+            .spawn(move || match actor.spawn_cargo_handle() {
+                Ok(cargo_handle) => actor.run(receiver_to_cancel, cargo_handle),
+                Err(_err) => {
+                    todo!()
+                }
+            })
+            .expect("failed to spawn thread");
```

Comment:
Na przyszłość - może to rozdzielić na new i spawn, pozwoliłoby to na runowanie w podobny sposób jak inne handlery. Wydaje mi się że można by tego nie tworzyć tutaj w nowym wątku, tylko zrobić normalną funkcje zwracająca response tak jak inne handlery.
Mogę ja się zając tą zmianą w jakimś osobnym PR później

Reply:
[Przedyskutowane na spotkaniu] Zostawiamy jako rzecz do zrobienia w przyszlosci